src/cmds/farm.py

```diff
@@ -0,0 +1,321 @@
+            print(f"Exception from 'full node' {e}")
+
+    client.close()
+    await client.await_closed()
+    return blockchain_state
+
+
+async def get_wallets_stats(wallet_rpc_port: int) -> Optional[Dict[str, Any]]:
+    stats = None
+    try:
+        config = load_config(DEFAULT_ROOT_PATH, "config.yaml")
+        self_hostname = config["self_hostname"]
+        if wallet_rpc_port is None:
+            wallet_rpc_port = config["wallet"]["rpc_port"]
+        wallet_client = await WalletRpcClient.create(self_hostname, uint16(wallet_rpc_port), DEFAULT_ROOT_PATH, config)
+        wallets = await wallet_client.get_wallets()
+        stats = compute_wallets_stats(wallets)
+    except Exception as e:
+        if isinstance(e, aiohttp.client_exceptions.ClientConnectorError):
+            print(f"Connection error. Check if wallet is running at {wallet_rpc_port}")
+        else:
+            print(f"Exception from 'wallet' {e}")
+
+    wallet_client.close()
+    await wallet_client.await_closed()
+    return stats
+
+
+async def is_farmer_running(farmer_rpc_port: int) -> bool:
+    is_running = False
+    try:
+        config = load_config(DEFAULT_ROOT_PATH, "config.yaml")
+        self_hostname = config["self_hostname"]
+        if farmer_rpc_port is None:
+            farmer_rpc_port = config["farmer"]["rpc_port"]
+        farmer_client = await FarmerRpcClient.create(self_hostname, uint16(farmer_rpc_port), DEFAULT_ROOT_PATH, config)
+        await farmer_client.get_connections()
+        is_running = True
+    except Exception as e:
+        if isinstance(e, aiohttp.client_exceptions.ClientConnectorError):
+            print(f"Connection error. Check if farmer is running at {farmer_rpc_port}")
+        else:
+            print(f"Exception from 'farmer' {e}")
+
+    farmer_client.close()
+    await farmer_client.await_closed()
+    return is_running
+
+
+async def get_challenges(farmer_rpc_port: int) -> Optional[Dict[str, Any]]:
+    signage_points = None
+    try:
+        config = load_config(DEFAULT_ROOT_PATH, "config.yaml")
+        self_hostname = config["self_hostname"]
+        if farmer_rpc_port is None:
+            farmer_rpc_port = config["farmer"]["rpc_port"]
+        farmer_client = await FarmerRpcClient.create(self_hostname, uint16(farmer_rpc_port), DEFAULT_ROOT_PATH, config)
+        signage_points = await farmer_client.get_signage_points()
+    except Exception as e:
+        if isinstance(e, aiohttp.client_exceptions.ClientConnectorError):
+            print(f"Connection error. Check if farmer is running at {farmer_rpc_port}")
+        else:
+            print(f"Exception from 'farmer' {e}")
+
+    farmer_client.close()
+    await farmer_client.await_closed()
+    return signage_points
+
+
+async def challenges(farmer_rpc_port: int) -> None:
+    signage_points = await get_challenges(farmer_rpc_port)
+    if signage_points is None:
+        return
+
+    signage_points.reverse()
+    for signage_point in signage_points:
```

Comment:
I don't know if we want to limit the number of challenges here, I can just add a `[:20]` here for example. Or I can allow to pass a --limit parameter with a default value of `X` and allow to pass 0 for now limit, or something like that. Any suggestion?

Reply:
a default of 20 and the ability to set a higher or lower one is the right way to go.